### PR TITLE
Adding more table config validation

### PR DESCRIPTION
**Pull request by icefury71**
## Description
Related to issue #5942 
* Adding more table config validation (retention, pushType and tenant)
* Fixing bug where retention manager does not work for real-time tables if the pushType is missing.

## Upgrade Notes
Does this PR prevent a zero down-time upgrade?
No

Does this PR fix a zero-downtime upgrade introduced earlier?
No

Does this PR otherwise need attention when creating release notes? 
No

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java [hunk 1]
```diff
@@ -76,10 +79,41 @@ public static void validate(TableConfig tableConfig, @Nullable Schema schema) {
...
+      Set<String> brokerTenants) {
+    validate(tableConfig, schema);
+    validateTenants(tableConfig, serverTenants, brokerTenants);
+  }
+
+  /**
+   * Ensures a valid tenant config is provided in the table config.
+   * Ensures the broker and server tenant names are valid.
+   */
+  private static void validateTenants(TableConfig tableConfig, Set<String> serverTenants, Set<String> brokerTenants) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java:96]: This is the same check as `PinotHelixResourceManager.validateTableTenantConfig(TableConfig tableConfig)`

**Review by Jackie-Jiang — COMMENTED** [review]
Recommend removing the tenant check in this PR, and move the validation into `PinotHelixResourceManager` in the following PR

**Comment by icefury71** [comment]
> Recommend removing the tenant check in this PR, and move the validation into `PinotHelixResourceManager` in the following PR

Done. Removed tenant validation

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 2]
```diff
@@ -1374,6 +1378,14 @@ private void assignInstances(TableConfig tableConfig, boolean override) {
...
   public void updateTableConfig(TableConfig tableConfig)
       throws IOException {
+    TenantConfig tenantConfig = tableConfig.getTenantConfig();
+    String brokerTag = tenantConfig.getBroker();
+    String serverTag = tenantConfig.getServer();
+    if (brokerTag == null || serverTag == null) {
+      if (!_isSingleTenantCluster) {
+        throw new InvalidTableConfigException("server and broker tenants must be specified for multi-tenant cluster");
+      }
+    }
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:1388]: You should put default tenant here if it is single-tenant cluster, similar to `addTable()`

You may extract this part into a helper method

Comment (icefury71) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:1388]: makes sense. lemme do that

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 3]
```diff
@@ -1195,12 +1182,26 @@ public void addTable(TableConfig tableConfig)
...
    */
   @VisibleForTesting
   void validateTableTenantConfig(TableConfig tableConfig) {
-    String tableNameWithType = tableConfig.getTableName();
     TenantConfig tenantConfig = tableConfig.getTenantConfig();
+    String brokerTag = tenantConfig.getBroker();
+    String serverTag = tenantConfig.getServer();
+    if (brokerTag == null || serverTag == null) {
+      if (!_isSingleTenantCluster) {
+        throw new InvalidTableConfigException("server and broker tenants must be specified for multi-tenant cluster");
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:1197]: That's also put the table name within the exception message

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

**Pull request merged**